### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Volosh1n**

### CODE REVIEW

File: lib/factory.rb [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+# - dig
+# - size/length
+# - members
+# - select
+# - to_a
+# - values_at
+# - ==, eql?
+
+class Factory
+  def self.new(*args, &block)
```
Comment (dzemlianoi-double) [inline lib/factory.rb]: class << self

File: lib/factory.rb [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      def each(&block)
+        values.each(&block)
+      end
+
+      def each_pair(&pair)
+        hash.each_pair(&pair)
+      end
+
+      def dig(*args)
+        hash.dig(*args)
```
Comment (dzemlianoi-double) [inline lib/factory.rb]: self wroten method dig

File: lib/factory.rb [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+      def initialize(*vars)
+        raise ArgumentError, "Expected #{args.count} arguments" if args.count != vars.count
+
+        args.each_index do |index|
+          instance_variable_set("@#{args[index]}", vars[index])
+        end
+      end
+
+      define_method :args do
```
Comment (dzemlianoi-double) [inline lib/factory.rb]: remove

File: lib/factory.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+        def each(&block)
+          values.each(&block)
+        end
+
+        def each_pair(&pair)
+          to_h.each_pair(&pair)
+        end
+
+        def dig(*args)
```
Comment (dzemlianoi-double) [inline lib/factory.rb:42]: don't use dig in dig method.

Look at reduce

File: lib/factory.rb [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          args.reduce(to_h) { |memo, arg| (memo[arg].is_a? NilClass) ? (return nil) : memo[arg] }
```
Comment (dzemlianoi-double) [inline lib/factory.rb]: don't copy not fully correct methods from other students repos.
https://github.com/WoLkErSs/Factory/pull/1/files#diff-404248b357b047b27fceb12341cece15R54

**Review by valikos — APPROVED** [review]

**Pull request merged**